### PR TITLE
Overhaul run comparisons

### LINKED ISSUE #263: Compare page should become a first-class citizen
This task originally had a smaller scope, but as I was writing it I realized what I'd actually like is a comparison page that is more of a first-class citizen in general.

Right now the comparison page is adaptable enough to compare any two splits, even those of different games! What would be great is a UI flow for comparing any two splits of a given game, or at least any two of your own splits. This would be similar to the "compare to best on record" or the rivals system.

Proposed flow:
1. Go to a splits page (probably PB)
2. Click "compare" link (would replace "compare to best on record" or "compare to my PB")
3. Be presented a list of splits associated with the same game [alias] and category [alias]. Best on record, your PB, and your last PB are all be stuck to the top in their own UI elements (or at least highlighted in some way). There is also a box available for entering any splits ID.
4. Select a set of splits against which to compare.
5. Be presented the current compare page, plus a multitude of stats and graphs below the splits timeline.

Background:
I recently had someone ask me if they could compare one PB of theirs to another on Splits I/O. I figured it was possible, but I had to go through the cumbersome process of opening both PBs to grab the IDs, and then adding "/<id1>/compare/<id2>" to the URL bar. I had also been bragging about the awesome stats and graphs provided on PBs and was a bit disappointed to find none on the comparison page. It would be awesome if there were some! :)

### LINKED ISSUE #499: Compare videos side-by-side
1. Comparing to another run should show both run's videos side-by-side, if present
2. Clicking the timeline should navigate both videos to that segment in the video, like it does on the current run page

### PR DESCRIPTION
**Pull request by glacials**
Overhaul run comparisons!

- Rolls the compare page into the run page (closes #263)
- Charts display both runs
- List of splits displays deltas
- Timelines show adjacent to each other (as before)
- Videos show side-by-side and seek together (closes #499)

However!

- Comparing two different runners is limited to tier-2 patrons, e.g.
  - Comparing your current PB to your previous PB is available to everyone
  - Comparing another runner's current PB to their previous PB is available to everyone
  - Comparing another runner's PB to your PB is limited to tier-2 patrons

Open to discussion from anyone about this last part, as comparisons already exist as a feature and I am generally against turning free features into a paid features.

However what exists today is so far below what's in this branch that calling it the same feature feels like an undersell. Nearly everything on the run page has been refactored to support either 2 or n runs, meaning future features on the run page should support comparisons natively, whereas the old comparison page was its own code that fell out of date as the run page was updated.

Check it out on [beta.splits.io](https://beta.splits.io/).

Screen Shot 2019-04-25 at 4 02 13 PM

beta splits io_oqt_compare=3v15 (1)

@BatedUrGonnaDie: Sorry this is so big :X

### CODE REVIEW

**Comment by jhobz** [comment]
Regarding the patron thing, what if everyone could do a simple timeline compare (and possibly the side-by-side video, since they could just load those up manually anyways) and then the rest of the page is locked behind patreon, like the graph on the run page?

File: app/views/runs/index.slim [hunk 1]
```diff
@@ -6,31 +6,30 @@
...
+              = current_user.runs.where(created_at: 1.year.ago..Time.now.utc).count
+          span.statcard-desc Runs stored
+        .col-sm-3: .statcard.p-3
+          h2.statcard-number
+            span> title=current_user.created_at = time_ago_in_words(current_user.created_at)
+          span.statcard-desc Account age
+      .card.shadow
+        = render partial: 'shared/run_table', locals: { \
+          runs: current_user.pbs, \
+          cols: %i[time name video uploaded owner_controls rival], \
```
Comment (BatedUrGonnaDie) [inline app/views/runs/index.slim]: Any reason to not add `:video` on the profile page as well?

Comment (glacials) [reply app/views/runs/index.slim]: Nope just forgot. Added.

File: app/views/runs/index.slim [hunk 2]
```diff
@@ -6,31 +6,30 @@
...
-            span.statcard-desc Runs stored
-          .col-sm-3: .statcard.p-3
-            h2.statcard-number
-              span> title=current_user.created_at = time_ago_in_words(current_user.created_at)
-            span.statcard-desc Account age
-        .card.shadow
-          = render partial: 'shared/run_table', locals: { \
-            runs: current_user.pbs, \
-            cols: %i[time name uploaded owner_controls rival], \
-            col_options: {time: [:archived]}, \
```
[the remark this answers is not in the document]
Comment (glacials) [reply app/views/runs/index.slim]: Oops, bad merge

**Comment by glacials** [comment]
@jhobz: Yeah I actually like this more, great idea! The code is much cleaner too. Gonna continue to clean some stuff up.

**Pull request merged**